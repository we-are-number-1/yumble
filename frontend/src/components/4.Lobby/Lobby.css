/*Lobby box */
.LobbyBox {
  outline: none;
  border: 4px solid #000000;
  background-color: white;
  height: auto;
  width: 75vw;
  font-family: Comic Sans MS;
  border-radius: 30px;
  padding: 2vw;
  font-size: 3vw;
  margin-bottom: 1vw;
  text-align: right;
}

.CusineTitle {
  margin-top: 10vw;
  margin-bottom: 1vw;
  font-weight: bold;
  color: white;
  font-family: Comic Sans MS;
  font-size: 2vw;
}

.ShareButton {
  cursor: pointer;
  outline: none;
  font-family: Comic Sans MS;
  font-size: 2vw;
  border-radius: 30px;
  border-color: black;
  border-width: 5px;
  padding: 0vw 1.5vw 0vw 1.5vw;
  background-color: yellow;
  position: absolute;
  top: 0;
  right: 0;
  margin: 2vw;
}

.ShareButton:hover {
  background: white;
}

.FoodIconText {
  font-size: 2.5vw;
  display: block;
  position: relative;
  top: 5.7vw;
  overflow: auto;
  white-space: nowrap;
  text-overflow: clip;
  max-width: auto;
}

.Inline_Block {
  display: inline-block;
}

.MakeTextCentre {
  text-align: center;
}

.CentreTitle {
  display: inline-block;
  width: 15.3vw;
}

/*Lobby image container*/
#container {
  width: 100%;
  text-align: center;
}
#container > div {
  width: calc(100% / 10);
  display: inline-block;
  vertical-align: top;
  border: 1px solid black;
  border-width: 0.15vw;
  text-align: center;
  margin: 2.8%;
  padding: 1.5vw;
}

#FoodIcon {
  background-position: center;
  background-size: cover;
}

.Food0 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRxoYX3yYvMaw2juMh5k-pQdZxi3iqi3Z_FZvvkkC0riaG9x0I2&s');
}
.Food1 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQYs1mLqODqg2dVakwOfQP3ekQxMUgNpjt5BMuOGXC3WgPbsDit&s');
}
.Food2 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSA8oOJICNTY5QwNpHakzDoCWf2pDYFeyGzTZ-ZVjXo70hNtjU&s');
}
.Food3 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEo8nBiDhQGhptSb2hhbWhKrawgLvpPX0q3DubgXgIX6NWqvw&s');
}
.Food4 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRwGzp3IpAdCzRuIOL7_kKZmlxgd-9AdpkYDOqFzJ7NSfaf6pYx&s');
}
.Food5 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQLSzQuKxwpH3K1wA5_OSZrkMJgYgXzDXaUzGSOHzyBQcapee8&s');
}
.Food6 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQlhMYQDLeaxAlB0NtpM6FNGrJP-tsWHPvkn9DxLC2wAsTV6Sk&s');
}
.Food7 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSqHcWePcAGjIQSJS-0fa5dyMd-i2XwgtL0fZO_JrrJn_ePvAg&s');
}
.Food8 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcREocbdf5t5MKIcglNisJaynTKmfZ0wJNUNxmJFJ1PT0kI95UT0&s');
}
.Food9 {
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSY5ltOFSPRyOab_JHIWnhqmBjI3ORCmfWMfM8wittlLrg_MBI&s');
}
