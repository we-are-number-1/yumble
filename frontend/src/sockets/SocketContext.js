import React, {useState} from 'react';
import {io} from 'socket.io-client';

const SocketContext = React.createContext();

/**
 * @param {*} param0
 * @return {*} SocketContext Provider
 */
function SocketContextProvider({children}) {
  const [socket, setSocket] = useState(() => {
    setSocket(io());
  });
  const [countdown, setCountdown] = useState(null);
  const [users, setUsers] = useState(null);
  const [preferences, setPreferences] = useState(null);

  const context = {
    socket,
    countdown,
    setCountdown,
    users,
    setUsers,
    preferences,
    setPreferences,
  };

  return (
    <SocketContext.Provider value={context}>
      {children}
    </SocketContext.Provider>
  );
}

export {
  SocketContext,
  SocketContextProvider,
};
