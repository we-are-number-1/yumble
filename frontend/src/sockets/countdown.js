/**
 * Function for processing game start countdown
 * @param {*} socket
 * @param {*} cb call back function
 */
export function countdown(socket, cb) {
  socket.on('countdown', ({count}) => {
    // Call back function with current count
    cb(count);
  });
}
