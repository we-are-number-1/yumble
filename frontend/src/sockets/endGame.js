/**
 * Function for when the game ends.
 * @param {*} socket
 * @param {*} cb call back function
 */
export function endGame(socket, cb) {
  socket.on('end_game', (data) => {
    socket.disconnect(true);
    cb(data);
  });
}
