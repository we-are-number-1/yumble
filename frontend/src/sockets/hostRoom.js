/**
 * Emits a host_room even to backend sockets
 * @param {*} socket
 * @param {*} sessionId
 */
export function hostRoom(socket, sessionId) {
  socket.emit('host_room', {sessionId});
}
