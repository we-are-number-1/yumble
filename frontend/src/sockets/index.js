export {getSocketClient} from './socketClient';
export {newUser} from './newUser';
