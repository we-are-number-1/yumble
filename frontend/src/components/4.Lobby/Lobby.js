import {useHistory, Redirect} from 'react-router-dom';
import React, {useState, useEffect, useContext} from 'react';
import Help from '../Common/Help';
import {SocketContext} from '../../sockets/SocketContext';
import * as SocketEvents from '../../sockets';
import '../Common/Help.css';
import './Lobby.css';

const Lobby = (props) => {
  const socketContext = useContext(SocketContext);
  const [ShareButtonPopup, setSharePopup] = useState(false);
  const [helpButtonPopup, setHelpButtonPopup] = useState(false);
  const [users, setUsers] = useState(
    socketContext.users ? socketContext.users : [],
  );
  const [redirect, setRedirect] = useState(false);
  const [cardData, setCardData] = useState(props.location.state);

  useEffect(() => {
    setUsers(socketContext.users ? socketContext.users : []);
  }, [socketContext]);

  useEffect(() => {
    document.title = 'Waiting Room';
    SocketEvents.newUser(socketContext.socket, (data) => {
      socketContext.setUsers(data.users);
    });
    SocketEvents.updateRestaurants(
        socketContext.socket,
        (data) => {
          if (data) {
            setCardData(data);
          }
        },
    );
    SocketEvents.countdown(socketContext.socket, (count) => {
      socketContext.setCountdown(count);
      startCountdown();
    });
  }, []);

  /**
   *
   */
  function startCountdown() {
    // go to next page
    setRedirect(true);
  }

  const history = useHistory();
  const goBack = () => {
    SocketEvents.leaveRoom(socketContext.socket);
    history.goBack();
  };

  let NumOfUsers = 0;
  NumOfUsers = users.length;

  const peopleList = () => {
    const peopleArray = [];
    const Food = 'Food';

    // Pushing users data into lobby dynamically
    for (let i = 0; i < NumOfUsers; i++) {
      const FoodID = Food.concat(i.toString());
      peopleArray.push(
          <div className={FoodID} id='FoodIcon' key={i.toString()}>
            <text className='FoodIconText'>{users[i]}</text>
          </div>,
      );
    }
    return peopleArray;
  };

  return (
    <>
      <h1 className='Title'>yumble</h1>
      <div className='MakeCentre'>
        <div className={'LobbyBox'}>
          <div>
            <div className='Inline_Block'>Group code: {socketContext.code}</div>
            <span className='CentreTitle'></span>
            <div className='Inline_Block'>{NumOfUsers}/10</div>
          </div>
          <div id='container'>{peopleList()}</div>
        </div>
        {socketContext.host && (
          <button
            className='GoButton'
            onClick={() =>
              SocketEvents.start(socketContext.socket, socketContext.code)
            }
          >
            Go
          </button>
        )}
        {redirect && (
          <Redirect to={{ pathname: '/CountDown', state: cardData }} />
        )}
        <button onClick={() => setSharePopup(true)} className='ShareButton'>
          Share
        </button>
        <Help trigger={ShareButtonPopup} setTrigger={setSharePopup}>
          <div className='MakeTextCentre'>
            <h2> Share the group code:</h2>
            <h1><b>{socketContext.code}</b></h1> 
            <a href={'https://yumble.xyz'}>
            https://yumble.xyz
            </a>
          </div>
        </Help>
        <button className='SmallBtn' id='BackButton' onClick={() => goBack()}>
          Back
        </button>
        <button
          onClick={() => setHelpButtonPopup(true)}
          className='SmallBtn'
          id='HelpButton'
        >
          help?
        </button>

        <Help trigger={helpButtonPopup} setTrigger={setHelpButtonPopup}>
          <p className='MakeTextCentre'>
            This is the lobby room, please wait a moment for the lobby to be
            filled.
          </p>
        </Help>
      </div>
    </>
  );
};

export default Lobby;
