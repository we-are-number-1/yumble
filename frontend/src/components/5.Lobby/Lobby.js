import {useHistory, Redirect} from 'react-router-dom';
import React, {useState, useEffect, useContext} from 'react';
import Help from '../Common/Help';
import {SocketContext} from './../../sockets/SocketContext';
import * as SocketEvents from './../../sockets';
import '../Common/Help.css';
import '../5.Lobby/Lobby.css';

const Lobby = () => {
  const socketContext = useContext(SocketContext);
  const [ShareButtonPopup, setSharePopup] = useState(false);
  const [helpButtonPopup, setHelpButtonPopup] = useState(false);
  const [users, setUsers] = useState(
    socketContext.users ? socketContext.users : []);
  const [redirect, setRedirect] = useState(false);

  useEffect(() => {
    setUsers(socketContext.users ? socketContext.users : []);
  }, [socketContext]);

  useEffect(() => {
    document.title = 'Waiting Room';
    SocketEvents.countdown(socketContext.socket, (count) => {
      console.log(count);
      socketContext.setCountdown(count);
      startCountdown();
    });
  }, []);


  /**
   *
   */
  function startCountdown() {
    // go to next page
    setRedirect(true);
  }

  const history = useHistory();
  const goBack = () => {
    SocketEvents.leaveRoom(socketContext.socket);
    history.goBack();
  };


  {
    /* link this to backend.
  Backend just needs to push data into
  People array, NumOfCusines array and GroupCode*/
  }
  let GroupCode = '';
  GroupCode = 'HX8192';

  const NumOfCusines = [];

  NumOfCusines.length == 0 ? NumOfCusines.push('Thai') : null;
  NumOfCusines.length > 0 ?

      (NumOfCusines.push(', ' + 'Burger'),
      NumOfCusines.push(', ' + 'European'),
      NumOfCusines.push(', ' + 'Mediterranean'),
      NumOfCusines.push(', ' + 'Chinese')) : null;


  let NumOfUsers = 0;
  NumOfUsers = users.length;

  const peopleList = () => {
    const peopleArray = [];
    const Food = 'Food';

    for (let i = 0; i < NumOfUsers; i++) {
      const FoodID = Food.concat(i.toString());
      peopleArray.push(
          <div className={FoodID} id='FoodIcon' key={i.toString()}>
            <text className='FoodIconText'>{users[i]}</text>
          </div>,
      );
    }
    return peopleArray;
  };


  return (
    <>
      <h1 className='Title'>yumble</h1>
      <div className='MakeCentre'>
        <div className='CusineTitle'>Cusines: {NumOfCusines}</div>
        <div className={'LobbyBox'}>
          <div>
            <div className='Inline_Block'>Group code: {GroupCode}</div>
            <span className='CentreTitle'></span>
            <div className='Inline_Block'>{NumOfUsers}/10</div>
          </div>
          <div id='container'>{peopleList()}</div>
        </div>
        <button className='GoButton'
          onClick={() => SocketEvents.start(socketContext.socket, 'test')}>
          Go
        </button>
        {redirect && <Redirect to='/CountDown' />}
        <button onClick={() => setSharePopup(true)} className='ShareButton'>
          Share
        </button>
        <Help trigger={ShareButtonPopup} setTrigger={setSharePopup}>
          <div className='MakeTextCentre'>
            <h2> Share the group code:</h2>
            <h1>{GroupCode}</h1>
            <a href={'https://yumble.xyz/JoinGroup/'}>
            https://yumble.xyz/JoinGroup
            </a>
          </div>
        </Help>
        <button className='SmallBtn' id='BackButton'
          onClick={() => goBack()}>
            Back
        </button>
        <button
          onClick={() => setHelpButtonPopup(true)}
          className='SmallBtn'
          id='HelpButton'
        >
          help?
        </button>

        <Help trigger={helpButtonPopup} setTrigger={setHelpButtonPopup}>
          <p className='MakeTextCentre'>
            This is the lobby room, please wait a moment for the lobby to be
            filled.
          </p>
        </Help>
      </div>
    </>
  );
};

export default Lobby;
