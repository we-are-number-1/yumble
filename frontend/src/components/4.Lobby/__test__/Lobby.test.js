import React from 'react';
import renderer from 'react-test-renderer';
import Lobby from '../Lobby';
import { SocketContextProvider } from '../../../sockets/SocketContext';
import { BrowserRouter } from 'react-router-dom';

jest.mock('./../../../sockets/SocketContext');
jest.mock('./../../../sockets');

test('Lobby snapshot testing', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SocketContextProvider>
          <Lobby location={location} />
        </SocketContextProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
