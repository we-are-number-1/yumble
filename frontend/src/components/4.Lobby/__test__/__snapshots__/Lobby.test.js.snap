// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby snapshot testing 1`] = `
Array [
  <h1
    className="Title"
  >
    yumble
  </h1>,
  <div
    className="MakeCentre"
  >
    <div
      className="LobbyBox"
    >
      <div>
        <div
          className="Inline_Block"
        >
          Group code: 
        </div>
        <span
          className="CentreTitle"
        />
        <div
          className="Inline_Block"
        >
          0
          /10
        </div>
      </div>
      <div
        id="container"
      />
    </div>
    <button
      className="ShareButton"
      onClick={[Function]}
    >
      Share
    </button>
    
    <button
      className="SmallBtn"
      id="BackButton"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="SmallBtn"
      id="HelpButton"
      onClick={[Function]}
    >
      help?
    </button>
    
  </div>,
]
`;
